autorest/adal/sender.go

```diff
@@ -0,0 +1,46 @@
+package adal
```

Comment:
The reason for duplicating this is to avoid circular dependencies between autorest and adal packages. The autorest should import adal for authentication.

The Sender could be moved in a small package which will be use by both packages.